File: _stats_build/statistics/wr_ao1r.rb
require_relative "abstract/ao_rounds"

class WrAo1r < AoRounds
  def initialize
    @title = "World record Ao1R (Average of 1 Round) history"
    @note = "Ao1R: the average from the only round in a competition (single-round events)."
    @table_header = { "Result" => :right, "Improvement" => :right, "Days" => :right, "Person" => :left, "Date" => :left, "Competition" => :left, "Details" => :left }
  end

  def round_count
    1
  end
end

File: core/packages/stats-build/src/statistics/wr_ao1r.ts
// NOTE: Ao1R——一场比赛恰好 1 轮时的 average
import { AoRounds } from '../core/ao_rounds.js';

export class WrAo1r extends AoRounds {
  constructor() {
    super();
    this.title = 'Ao1R';
    this.titleZh = 'Ao1R';
    this.note = 'Average of 1 Round: a person\'s average when they competed in exactly 1 round at a competition.';
    this.noteZh = '单轮平均：一场比赛中恰好只参加 1 轮时的平均成绩。';
  }
  roundCount() { return 1; }
}
